framework/decode/dx12_json_consumer_base.cpp

```diff
@@ -0,0 +1,243 @@
+    FieldToJson(jdata["triangles_vertex_format"], data.triangles_vertex_format, options);
+    FieldToJson(jdata["triangles_index_count"], data.triangles_index_count, options);
+    FieldToJson(jdata["triangles_vertex_count"], data.triangles_vertex_count, options);
+    FieldToJson(jdata["triangles_vertex_stride"], data.triangles_vertex_stride, options);
+}
+
+static void
+FieldToJson(nlohmann::ordered_json& jdata, const format::DxgiAdapterDesc& data, const util::JsonOptions& options)
+{
+    FieldToJson(jdata["Description"], std::wstring_view(data.Description), options);
+    FieldToJson(jdata["VendorId"], data.VendorId, options);
+    FieldToJson(jdata["DeviceId"], data.DeviceId, options);
+    FieldToJson(jdata["SubSysId"], data.SubSysId, options);
+    FieldToJson(jdata["Revision"], data.Revision, options);
+    FieldToJson(jdata["DedicatedVideoMemory"], data.DedicatedVideoMemory, options);
+    FieldToJson(jdata["DedicatedSystemMemory"], data.DedicatedSystemMemory, options);
+    FieldToJson(jdata["SharedSystemMemory"], data.SharedSystemMemory, options);
+    FieldToJson(jdata["LuidLowPart"], data.LuidLowPart, options);
+    FieldToJson(jdata["LuidHighPar"], data.LuidHighPart, options);
+    // Should we break out the packed data? (2 bits (LSB) to store Type and 30 bits for object ID)
+    FieldToJson(jdata["extra_info"], data.extra_info, options);
+}
+
+static void
+FieldToJson(nlohmann::ordered_json& jdata, const format::Dx12RuntimeInfo& data, const util::JsonOptions& options)
+{
+    FieldToJson(jdata["version"], data.version, util::filepath::kFileVersionSize, options);
+    FieldToJson(jdata["src"], util::ViewOfCharArray(data.src, util::filepath::kMaxFilePropertySize), options);
+}
+
+static void
+FieldToJson(nlohmann::ordered_json& jdata, const util::filepath::FileInfo& data, const util::JsonOptions& options)
+{
+    FieldToJson(jdata["ProductVersion"], ViewOfCharArray(data.ProductVersion, filepath::kMaxFilePropertySize), options);
+    FieldToJson(jdata["FileVersion"], ViewOfCharArray(data.FileVersion, filepath::kMaxFilePropertySize), options);
+    FieldToJson(jdata["AppVersion"], data.AppVersion, filepath::kMaxFilePropertySize, options);
+    FieldToJson(jdata["AppName"], ViewOfCharArray(data.AppName, filepath::kMaxFilePropertySize), options);
+    FieldToJson(jdata["CompanyName"], ViewOfCharArray(data.CompanyName, filepath::kMaxFilePropertySize), options);
+    FieldToJson(
+        jdata["FileDescription"], ViewOfCharArray(data.FileDescription, filepath::kMaxFilePropertySize), options);
+    FieldToJson(jdata["InternalName"], ViewOfCharArray(data.InternalName, filepath::kMaxFilePropertySize), options);
+    FieldToJson(
+        jdata["OriginalFilename"], ViewOfCharArray(data.OriginalFilename, filepath::kMaxFilePropertySize), options);
+    FieldToJson(jdata["ProductName"], ViewOfCharArray(data.ProductName, filepath::kMaxFilePropertySize), options);
+}
+
+GFXRECON_END_NAMESPACE(util)
```

Comment:
Perhaps these could go in the now-empty [framework/decode/dx12_decode_json_util.h](https://github.com/LunarG/gfxreconstruct/pull/1362/files#diff-b07efe17028ec7db930495680666e254d3803c87a73d03b47b029e1d77fee3a0), although they are currently only used in this one compilation unit here.